R/insertPackage.R

```diff
@@ -56,41 +59,59 @@ insertPackage <- function(file,
         setwd(curwd)
     }
     
-    ## TODO: maybe branch on Windows / OS X files
-    srcdir <- file.path(repodir, "src", "contrib")
-    if (!file.exists(srcdir)) stop("Directory ", srcdir, " not found\n", .Call=FALSE)
-
-    ## copy file into repo
-    if (!file.copy(file, srcdir, overwrite=TRUE)) {
-        stop("File ", file, " can not be copied to ", srcdir, .Call=FALSE)
-    }
+    fileext <- tools::file_ext(file) 
     
-    ## update index
-    write_PACKAGES(srcdir, type="source")
-    ## TODO: generalize to binary
+    if (fileext == "gz"){
```

Comment:
This also is not quite good enough. I borrowed a reg,exp. from R's tools package, see the pruneRepo.R file:

      ext <- "_.*\\.tar\\..*$"            # with a nod to src/library/tools/packages.R